### PR TITLE
[FIX] Fix value_list_validator type deduction for char.

### LINKED ISSUE #1545: value_list_validator{"foobar"} deduces to value_list_validator<double>
... instead of to `value_list_validator<std::string>`, because `char` satisfies `arithmetic`. `char` should be excluded from arithmetic *for the deduction guide*.

### PR DESCRIPTION
**Pull request by smehringer**
Fixes #1545

### CODE REVIEW

File: include/seqan3/argument_parser/validators.hpp [hunk 1]
```diff
@@ -264,15 +264,25 @@ class value_list_validator
  */
 //!\brief Deduction guide for a parameter pack over an arithmetic type.
 template <arithmetic ...option_types>
+//!\cond
+    requires !(std::same_as<char, option_types> || ...)
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: it should 
```suggestion
    requires !(seqan3::detail::is_char_adaptation_v<option_types> || ...)
```

Actually didn't we have a char concept?

File: include/seqan3/argument_parser/validators.hpp [hunk 2]
```diff
@@ -264,15 +264,25 @@ class value_list_validator
...
+//!\endcond
 value_list_validator(option_types...) -> value_list_validator<double>;
 
 //!\brief Deduction guide for ranges over an arithmetic type.
 template <std::ranges::forward_range range_type>
 //!\cond
-    requires arithmetic<std::ranges::range_value_t<range_type>>
+    requires arithmetic<std::ranges::range_value_t<range_type>> &&
+             !std::same_as<std::ranges::range_value_t<range_type>, char>
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: same here

File: include/seqan3/argument_parser/validators.hpp [hunk 3]
```diff
@@ -264,15 +264,25 @@ class value_list_validator
...
 //!\endcond
 value_list_validator(range_type && rng) -> value_list_validator<double>;
 
+//!\brief C-style string deduces to std::string.
+value_list_validator(char const *) -> value_list_validator<std::string>;
+
+//!\brief C-style string deduces to std::string.
+value_list_validator(char *) -> value_list_validator<std::string>;
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]: why was the one below not picked up?

Comment (smehringer) [reply include/seqan3/argument_parser/validators.hpp]: I don't know? Because parameter packs cannot capture a char pointer into a pack of chars?

**Comment by smehringer** [comment]
@rrahn polite ping

File: include/seqan3/argument_parser/validators.hpp (hunk 1 above)
Comment (smehringer) [reply include/seqan3/argument_parser/validators.hpp]: I cannot find a concept. Just the templetized variable

**Comment by smehringer** [comment]
@rrahn polite ping

File: include/seqan3/argument_parser/validators.hpp [hunk 4]
```diff
@@ -264,26 +264,32 @@ class value_list_validator
...
 //!\endcond
 value_list_validator(option_types...) -> value_list_validator<std::string>;
 
 //!\brief Deduction guide for ranges over a value type convertible to std::string.
-template <std::ranges::forward_range range_type>
+template <typename range_type>
 //!\cond
-    requires std::constructible_from<std::string, std::ranges::range_value_t<range_type>>
+    requires (std::ranges::forward_range<std::decay_t<range_type>> &&
+                 std::constructible_from<std::string, std::ranges::range_value_t<range_type>>)
```
Comment (rrahn) [inline include/seqan3/argument_parser/validators.hpp]:
```suggestion
              std::constructible_from<std::string, std::ranges::range_value_t<range_type>>)
```

**Comment by smehringer** [comment]
Sure. It would be great if you have a few minutes this week because I am not sure if I understood the decay entirely :woman_student: 

PS: Make sure to squash the commits when merging!

**Comment by rrahn** [comment]
sure

**Comment by smehringer** [comment]
@rrahn Can you still merge this already?

**Pull request merged**